### PR TITLE
🆕 ArrayStack Update

### PR DESCRIPTION
**Pull request by jameshughes89**
### What
Update the `ArrayStack` class to include `equals` and `hash`.
Include the better test class as `ArrayStackTest2`. 

### Why
It's better. 

### Testing
Yersh. 

### Additional Notes
1. I will add a download link to the new tests to Topic6. 
2. I commented out the equals tests and import since it will mess with them too much.

### CODE REVIEW

File: src/main/java/ArrayStack.java [hunk 1]
```diff
@@ -88,4 +90,23 @@ public String toString() {
...
+        return top == that.top
+                && Arrays.equals(this.stack, 0, this.top, that.stack, 0, that.top);
+    }
+
+    @Override
+    public final int hashCode() {
+        int result = Objects.hash(top);
+        result = 31 * result;
+        for (int i = 0; i < Math.min(this.top, this.stack.length); i++){
+            result += stack[i].hashCode();
```
Comment (twentylemon) [inline src/main/java/ArrayStack.java:108]: Still not a great hashcode. Simply adding values leads to high collision rates. It's not a big deal for this I guess, but like. Ya know.

File: src/test/java/ArrayStackTest2.java [hunk 2]
```diff
@@ -0,0 +1,243 @@
+import org.junit.jupiter.api.*;
+
+import java.util.NoSuchElementException;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+
+class ArrayStackTest2 {
```
Comment (twentylemon) [inline src/test/java/ArrayStackTest2.java:8]: Why 2? Give it a name.

File: src/test/java/ArrayStackTest2.java [hunk 3]
```diff
@@ -0,0 +1,243 @@
...
+
+    private Stack<Integer> classUnderTest;
+    private Stack<Integer> preState;
+
+//    @Test
```
Comment (twentylemon) [inline src/test/java/ArrayStackTest2.java:13]: Just remove imo. Think it's confusing to leave it commented out. We have tests in the assignments repo, so we're confident enough the implementation is correct.

File: src/test/java/ArrayStackTest2.java [hunk 4]
```diff
@@ -0,0 +1,243 @@
...
+        }
+    }
+
+    @Nested
+    class WhenLarge{
+
+        @Test
+        @DisplayName("Pushing beyond capacity automatically calls expandCapacity.")
+        void push_large_successfullyExpandsCapacity(){
+            // Default capacity is 10
```
Comment (twentylemon) [inline src/test/java/ArrayStackTest2.java:236]: Is it 10?

Comment (jameshughes89) [reply src/test/java/ArrayStackTest2.java:236]: fek

File: src/test/java/ArrayStackTestThorough.java [hunk 5]
```diff
@@ -0,0 +1,193 @@
+import org.junit.jupiter.api.*;
+
+import java.util.NoSuchElementException;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+
+class ArrayStackTestThorough {
```
Comment (twentylemon) [inline src/test/java/ArrayStackTestThorough.java:8]: Test classes need to end with `Test`

**Review by twentylemon — APPROVED** [review]

**Pull request merged**